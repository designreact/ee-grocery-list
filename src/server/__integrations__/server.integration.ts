import request from 'supertest';
import { app, server } from '../index';

jest.mock('uuidv4', () => ({
  uuid: () => '0123456789',
}));

jest.mock('../utils/database');

afterAll(() => {
  server.close();
});

test('Client middleware works', async () => {
  const response = await request(app.callback()).get('/').send();
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
});

test('Auth middleware sets a cookie', async () => {
  const response = await request(app.callback()).get('/').send();
  expect(response.status).toBe(200);
  expect(response.res.headers['set-cookie']).toMatchSnapshot();
  expect(response.text).toMatchSnapshot();
});

describe('Items middleware', () => {
  // TODO: use configured local database to simulate add, remove & update
  test.skip('gets items', async () => {
    const response = await request(app.callback())
    .get('/items')
    .set('Cookie', ['userId=0123456789'])
    .send();
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('add an item', async () => {
    const response = await request(app.callback())
      .post('/items')
      .set('Cookie', ['userId=0123456789'])
      .send({
        action: 'add',
        item: {
          text: 'hello world'
        }
      });
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('update an item', async () => {
    const response = await request(app.callback())
      .post('/items')
      .set('Cookie', ['userId=0123456789'])
      .send({
        action: 'update',
        item: {
          id: 'item-id',
          text: 'hello world',
          checked: true,
        }
      });
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('delete an item', async () => {
    const response = await request(app.callback())
      .post('/items')
      .set('Cookie', ['userId=0123456789'])
      .send({
        action: 'delete',
        item: {
          id: 'item-id'
        }
      });
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
});
