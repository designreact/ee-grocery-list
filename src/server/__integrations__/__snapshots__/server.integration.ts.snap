// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth middleware sets a cookie 1`] = `
Array [
  "userId=0123456789; path=/; httponly",
]
`;

exports[`Auth middleware sets a cookie 2`] = `
"<!DOCTYPE html>
<html>
<head>
  <meta charset='utf-8'>
  <meta http-equiv='X-UA-Compatible' content='IE=edge'>
  <title>React Shopping List</title>
  <meta name='viewport' content='width=device-width, initial-scale=1'>
  <link rel='stylesheet' type='text/css' media='screen' href='main.css'>
</head>
<body>
  <div id=\\"root\\"></div>
</body>
</html>"
`;

exports[`Client middleware works 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <meta charset='utf-8'>
  <meta http-equiv='X-UA-Compatible' content='IE=edge'>
  <title>React Shopping List</title>
  <meta name='viewport' content='width=device-width, initial-scale=1'>
  <link rel='stylesheet' type='text/css' media='screen' href='main.css'>
</head>
<body>
  <div id=\\"root\\"></div>
</body>
</html>"
`;

exports[`Items middleware add an item 1`] = `"{\\"status\\":\\"ok\\",\\"items\\":[{\\"id\\":\\"itemid-0123456789\\",\\"text\\":\\"items with added\\",\\"checked\\":true,\\"userId\\":\\"userId-0987654321\\"}]}"`;

exports[`Items middleware delete an item 1`] = `"{\\"status\\":\\"ok\\",\\"items\\":[{\\"id\\":\\"itemid-0123456789\\",\\"text\\":\\"items with deleted\\",\\"checked\\":true,\\"userId\\":\\"userId-0987654321\\"}]}"`;

exports[`Items middleware gets items 1`] = `"{\\"status\\":\\"ok\\",\\"items\\":[{\\"id\\":\\"itemid-0123456789\\",\\"text\\":\\"items\\",\\"checked\\":true,\\"userId\\":\\"userId-0987654321\\"}]}"`;

exports[`Items middleware update an item 1`] = `"{\\"status\\":\\"ok\\",\\"items\\":[{\\"id\\":\\"itemid-0123456789\\",\\"text\\":\\"items with updated\\",\\"checked\\":true,\\"userId\\":\\"userId-0987654321\\"}]}"`;
